### PR TITLE
MON-15830 add invalid_on_negative attribute

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

**Fixes** # (issue)

when notification_number is greater than 2^15 in bbdo2.0 it's converted into negative value. So the value is stored in null value in database

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.10.x
- [X] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/resources/db_variables.robot [hunk 1]
```diff
@@ -3,7 +3,7 @@ ${EtcRoot}  /tmp/etc
 ${VarRoot}  /tmp/var
 ${DBName}	centreon_storage
 ${DBNameConf}	centreon
-${DBHost}	localhost
+${DBHost}	127.0.0.1
```
Comment (bouda1) [inline tests/resources/db_variables.robot:6]: I don't understand this change.
localhost => unix socket
127.0.0.1 => tcp socket

File: tests/broker-engine/output-tables.robot [hunk 2]
```diff
@@ -73,3 +74,43 @@ BEHS1
...
+	${result}=	Check Service Status With Timeout	host_16	service_314	2	30
+	Should Be True	${result}	msg=The service (host_16,service_314) is not CRITICAL as expected
+
+	Connect To Database	pymysql	${DBName}	${DBUser}	${DBPass}	${DBHost}	${DBPort}
+	${output}=	Query	SELECT s.notification_number FROM services s LEFT JOIN hosts h ON s.host_id=h.host_id WHERE h.name='host_16' AND s.description='service_314'
+	Should Be True	${output[0][0]} == None  msg=notification_number is not null
+
+
+
+	Sleep  5s
```
Comment (bouda1) [inline tests/broker-engine/output-tables.robot:112]: Why?

**Review by bouda1 — APPROVED** [review]

File: tests/resources/db_variables.robot (hunk 1 above)
Comment (jean-christophe81) [reply tests/resources/db_variables.robot:6]: used to connect to mariadb container

File: tests/broker-engine/output-tables.robot (hunk 2 above)
Comment (jean-christophe81) [reply tests/broker-engine/output-tables.robot]: forget

**Pull request merged**